File: Code_Test/py/dfs_bfs/5-1stack.py
# First in First out (stack on the top)
stack = []
stack.append(5)
stack.append(2)
stack.append(3)
stack.append(7)
stack.pop() # delete 7
stack.append(1)
stack.append(4)
stack.pop() # delete 4

print(stack)
print(stack[::-1])

File: Code_Test/cpp/dfs_bfs/5-1stack.cpp
#include <iostream>
#include <stack>
// LIFO (stack on the top of the tower)
std::stack<int> s; // how to use stack

int main(){
    // 삽입(5) - 삽입(2) - 삽입(3) - 삽입(7) - 삭제() - 삽입(1) - 삽입(4) - 삭제()
    s.push(5);
    s.push(2);
    s.push(3);
    s.push(7);
    s.pop();
    s.push(1);
    s.push(4);
    s.pop();

    while (!s.empty()){
        std::cout<<s.top()<<' '; // print the top element
        s.pop(); // delete the top element
    }
}
